esp-wifi/src/lib.rs

```diff
@@ -230,35 +230,6 @@ pub fn initialize(
     radio_clocks: hal::peripherals::RADIO_CLK,
     clocks: &Clocks,
 ) -> Result<EspWifiInitialization, InitializationError> {
-    #[cfg(any(esp32, esp32s3, esp32s2))]
-    const MAX_CLOCK: u32 = 240;
-
-    #[cfg(any(esp32c3, esp32c6))]
-    const MAX_CLOCK: u32 = 160;
-
-    #[cfg(esp32c2)]
-    const MAX_CLOCK: u32 = 120;
-
-    #[cfg(esp32h2)]
-    const MAX_CLOCK: u32 = 96;
-
-    if clocks.cpu_clock != MegahertzU32::MHz(MAX_CLOCK) {
```

Comment:
IIRC we still need some check to ensure that CPU frequency is greater that 80mhz

Reply:
I'm not entirely sure how this came to be in the first place. I tried tracing the origin of this code and I only found this place https://github.com/gemesa/esp32-phantom/blob/main/esp-wifi-promiscuous.patch

Reply:
I do realize now that different boards have different minimum clocks ([here](https://github.com/esp-rs/esp-hal/blob/d34a872230a214e3e7a4366e45675c72674a0a68/esp-hal/src/clock/mod.rs#L117-L118)) but before over-engineering anything I'd like to believe anything above 80MHz should be fine.

Also, is this the only way people consume the HAL can configure the clock? Cause if this is the case, there's no way to choose a CPU frequency below the minimum supported freq, that would make this check redundant. But if there are ways to circumvent that, the check makes sense.

On a side note, what source of truth do you usually check to verify those things?

Reply:
I'd say just having a check for > 80MHz should be included. While there is currently no way to configure less maybe we can in future and most probably will forget about the assumption here

Reply:
Should be hopefully resolved now.